loopdb/sql_store.go

```diff
@@ -0,0 +1,722 @@
+package loopdb
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/btcsuite/btcd/btcutil"
+	"github.com/btcsuite/btcd/chaincfg/chainhash"
+	"github.com/lightninglabs/loop/loopdb/sqlc"
+	"github.com/lightningnetwork/lnd/keychain"
+	"github.com/lightningnetwork/lnd/lntypes"
+	"github.com/lightningnetwork/lnd/routing/route"
+)
+
+// FetchLoopOutSwaps returns all swaps currently in the store.
+func (s *BaseDB) FetchLoopOutSwaps(ctx context.Context) ([]*LoopOut,
+	error) {
+
+	var loopOuts []*LoopOut
+
+	err := s.ExecTx(ctx, NewSqlReadOpts(), func(*sqlc.Queries) error {
+		swaps, err := s.Queries.GetLoopOutSwaps(ctx)
+		if err != nil {
+			return err
+		}
+
+		loopOuts = make([]*LoopOut, 0, len(swaps))
+
+		for _, swap := range swaps {
```

Comment:
nit: not causing any issues right now, but would be nicer to avoid this pattern and use an indexed for loop instead, since we're appending a converted loopvar to the `loopOuts` slice, if we ever end up adding a byte slice to the contract then it may cause problems. Similar comment for `FetchLoopIns`.